### ISSUE TITLE
Project policies

### ISSUE THREAD

**Issue by matthijskooijman** [projecthamster/hamster#585]
With our new maintainer team, I think it would be good to think a bit on how to approach this, and what policies we should adapt. This was already previously discussed a bit, so here's a proposal for working with issues and PRs:
 - All changes should go through PRs. Direct (and certainly force) pushes to master are not allowed (and also technically prevented by branch protection).
 - A PR can be merged when it has two accepting reviews. This includes the person doing the merge, but can also include reviews from community members without repo write access. For small or trivial changes, just one review can be sufficient. Also, for a PR from the maintainer team, just one review can also be sufficient (to allow a PR to be merged with just two maintainers involved).
 - PRs should have clear commit messages and commits should be clean and atomic (not mixing up different changes in the same commit). Rebasing and history editing of PRs is encouraged to keep the history clean (but sometimes *adding* fixup commits first and then squashing commits just before merge can be easier for review).
 - Issues should mostly be about bugs and feature requests and should remain open as long as the bug is present or there is intention to (someday) implement a feature. Issues that ask for help with a problem or ask questions are ok, but should usually be closed rather quickly (discussion can still continue on a closed issue anyway).
 - Issues that are previously reported, should be closed with a comment saying they're duplicate and pointing to the previous comment.

Something like that?

**Comment by mwilck** [comment]
ACK for all items. Thanks for writing this up.

Wrt the commit messages, you are introducing a culture change here. ederag's commit messages used to be terse to the extreme. But I agree it's a change for the better. We just have to make sure we don't exaggerate.

**Comment by ederag** [comment]
> ederag's commit messages used to be terse to the extreme

Indeed, information was given as comments/docstrings in the code itself.

**Comment by mwilck** [comment]
good to know that you're still listening :-)

**Comment by matthijskooijman** [comment]
I added some more details on handling PRs and added some links for extra documentation.

> Wrt the commit messages, you are introducing a culture change here. ederag's commit messages used to be terse to the extreme. But I agree it's a change for the better. We just have to make sure we don't exaggerate.

It's a change, but I'm not sure it is too big. I think @ederag has rather concise style of commit messages, and my personal style is quite verbose, but I would not want to impose my verbosity on the project (above, I only wrote "clear" wrt commit messages).

Wrt to clean and atomic commits, this is already the culture of this repo - @ederag has done a great job of cleanly separating unrelated changes and refactors from actual fixes.

**Comment by ederag** [comment]
Also, `git gui` is very handy to commit only part of the changes.
(add/remove lines from commit)

**Comment by ederag** [comment]
@matthijskooijman 
> For small (single commit) pullrequests, the "squash and merge" feature from github can be used to produce a single (non-merge) commit that includes the commit message and a reference to the pullrequest.

Here is why I preferred normal merge commits for all PR, even single commit ones.
With "squash and merge", there is no indication that the PR has been merged in `gitk`:

![Screenshot_20200318_211946](https://user-images.githubusercontent.com/10962809/77004096-d505f080-695e-11ea-8c71-95eb14df450e.png)

Just sharing; your call.

**Comment by taniwallach** [comment]
Maybe a summary of this could be moved to the Wki?
